zk/src/org/zkoss/zk/ui/impl/PageImpl.java

```diff
@@ -749,11 +749,10 @@ public void preInit() {
 	}
 
 	public void destroy() {
-		super.destroy();
-
 		for (Component root: this.getRoots()) {
-			root.detach();
+			getDesktop().getWebApp().getConfiguration().invokeDestroy(root);
```

Comment:
1. Be aware of NPE of chaining getter
2. Why not just call `afterComponentDetached(root, null);`